pkg/db/models/release_tag.go

```diff
@@ -75,3 +75,49 @@ func GetLastAcceptedByArchitectureAndStream(db *gorm.DB, release string) ([]Rele
 
 	return results, nil
 }
+
+// GetLastPayloadStatus returns the most recent payload status for an architecture/stream combination,
+// as well as the count of how many of the last payloads had that status (e.g., when this returns
+// Rejected, 5 -- it means the last 5 payloads were rejected.
+func GetLastPayloadStatus(db *gorm.DB, architecture, stream, release string) (string, int, error) {
+	count := struct {
+		Phase string `gorm:"column:phase"`
+		Count int    `gorm:"column:count"`
+	}{}
+
+	result := db.Raw(`
+		WITH releases AS
+			(
+				SELECT
+					ROW_NUMBER() OVER(ORDER BY "releaseTime" desc) AS id,
+					phase
+				FROM
+					release_tags
+				WHERE
+					architecture = ? AND stream = ? AND release = ?
+			),
+		changes AS
+			(
+				SELECT
+					*,
+					CASE WHEN LAG(phase) OVER(ORDER BY id) = phase THEN 0 ELSE 1 END AS change
+				FROM
+					releases
+			),
+		groups AS
+			(
+				SELECT
+					*,
+					SUM(change) OVER(ORDER BY id) AS group FROM changes
+			)
+		SELECT
+			phase, COUNT(phase)
+		FROM
+			groups
+		WHERE
+			groups.group = 1
+		GROUP BY
+			phase`, architecture, stream, release).Scan(&count)
```

Comment:
This absurd SQL statement brought to you by cobbling together several StackOverflow:tm: posts.  

It uses multiple [CTE](https://www.geeksforgeeks.org/cte-in-sql/) to group islands of release tags in a particular phase, and gets the most recent island. It lets us know the status (i.e. accepted or rejected) of the most recent payload, and also how many of them occurred in a row.